satellite/satellitedb/bandwidthagreement.go

```diff
@@ -8,20 +8,35 @@ import (
 	"time"
 
 	"storj.io/storj/pkg/bwagreement"
+	"storj.io/storj/pkg/utils"
 	dbx "storj.io/storj/satellite/satellitedb/dbx"
 )
 
 type bandwidthagreement struct {
 	db *dbx.DB
 }
 
-func (b *bandwidthagreement) CreateAgreement(ctx context.Context, agreement bwagreement.Agreement) error {
-	_, err := b.db.Create_Bwagreement(
-		ctx,
-		dbx.Bwagreement_Signature(agreement.Signature),
-		dbx.Bwagreement_Data(agreement.Agreement),
-	)
-	return err
+func (b *bandwidthagreement) CreateAgreement(ctx context.Context, serialNum string, agreement bwagreement.Agreement) error {
+	tx, err := b.db.Open(ctx)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	_, err = tx.Get_Bwagreement_By_Serialnum(ctx, dbx.Bwagreement_Serialnum(serialNum))
```

Comment:
I don't think we need the serialNum query and transaction. The serial number field is defined as `unique`, so an error is returned if we attempt to enter an agreement with a pre-existing serial number